meta-mender-core/recipes-bsp/u-boot/files/uboot_auto_patch.sh

```diff
@@ -0,0 +1,359 @@
+#!/bin/bash
+
+if [ -z "$1" ] || [ "$1" = "-h" ] || [ "$1" = "--help" ]; then
+    echo "This script should not be run on its own. Use uboot_auto_configure.sh."
+    exit 1
+fi
+
+set -e
+
+get_candidate_list() {
+    # Search for all files that are relative paths, and do not start with what
+    # looks like an option flag (-I), and return the list.
+    for file in $(egrep -io '(^| )([a-z0-9][-.a-z0-9_/]*)?include[-.a-z0-9_/]*' "$DEP_FILE"); do
+        if [ ! -f "$file" ]; then
+            continue
+        fi
+
+        # Don't change env_default.h or any mender file, we have those under
+        # control.
+        bname="$(basename "$file")"
+        if [ "$bname" = env_default.h ] || grep -q 'mender' <<<"$bname"; then
+            continue
+        fi
+
+        echo "$file"
+    done
+}
+
+patch_candidate_list() {
+    # Execute supplied sed command on whole candidate list.
+    for file in $(get_candidate_list); do
+        sed -i -rne "$1" "$file"
+    done
+}
+
+definition_for_c_files() {
+    if [ -n "${2:-}" ]; then
+        echo "#define $1 $2"
+    elif [ -n "${1:-}" ]; then
+        echo "#define $1"
+    else
+        echo ""
+    fi
+}
+
+definition_for_kconfig_files() {
+    if [ -n "${2:-}" ]; then
+        echo "$1=$2"
+    elif [ -n "${1:-}" ]; then
+        echo "$1=y"
+    else
+        echo ""
+    fi
+}
+
+replace_definition() {
+    # Replaces definitions in source files, taking into account that it may span
+    # multiple lines using '\'.
+    # If the function is given only one argument, it deletes the definition, if
+    # given two, it replaces it with that definition, if given three, it gives
+    # the new definition that value.
+
+    c_repl="$(definition_for_c_files "${2:-}" "${3:-}")"
+    kconfig_repl="$(definition_for_kconfig_files "${2:-}" "${3:-}")"
+
+    # c_repl refers to replacement text in C files.
+    # c_sed_op refers to operation on C files.
+    # kconfig_repl refers to replacement text in Kbuild files.
+    if [ -n "${3:-}" ]; then
+        # Replace.
+        c_sed_op="s%.*%$c_repl%"
+        should_exist=1
+    elif [ -n "${2:-}" ]; then
```

Comment:
Isn't this exactly as the condition above? Can't you do `if [ -n "${3:-}" ] || [ -n "${2:-}" ]  ; then`?

Reply:
Yes, you're right, well spotted! Will fix.